### PR TITLE
AB#65902 Added Pages to create Scheme

### PR DESCRIPTION
**Pull request by Antek012**

### CODE REVIEW

File: src/HE.Investment.AHP.Domain/DomainExceptionHandler.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    {
+        try
+        {
+            return await action();
+        }
+        catch (DomainValidationException domainValidationException)
+        {
+            _logger.LogWarning(domainValidationException, "Validation error(s) occured: {Message}", domainValidationException.Message);
+
+            return new OperationResult<TResult>(domainValidationException.OperationResult.Errors, default!);
```
Comment (Carq) [inline src/HE.Investment.AHP.Domain/DomainExceptionHandler.cs:26]: `!` should not be used without very good reason.

File: src/HE.Investment.AHP.Domain/Scheme/CommandHandlers/CreateSchemeCommandHandler.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    public CreateSchemeCommandHandler(ISchemeRepository repository, IDomainExceptionHandler domainExceptionHandler)
+    {
+        _repository = repository;
+        _domainExceptionHandler = domainExceptionHandler;
+    }
+
+    public async Task<OperationResult<SchemeId>> Handle(CreateSchemeCommand request, CancellationToken cancellationToken)
+    {
+        return await _domainExceptionHandler.Handle(async () =>
```
Comment (Carq) [inline src/HE.Investment.AHP.Domain/Scheme/CommandHandlers/CreateSchemeCommandHandler.cs:22]: The better option is to use a Decorator for all Command handlers. We need to discuss it online.

File: src/HE.Investment.AHP.Domain/Scheme/CommandHandlers/CreateSchemeCommandHandler.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
+using HE.Investment.AHP.Domain.Scheme.Commands;
+using HE.Investment.AHP.Domain.Scheme.Entities;
+using HE.Investment.AHP.Domain.Scheme.ValueObjects;
+using HE.InvestmentLoans.Common.Validation;
+using MediatR;
+
+namespace HE.Investment.AHP.Domain.Scheme.CommandHandlers;
+
+public class CreateSchemeCommandHandler : IRequestHandler<CreateSchemeCommand, OperationResult<SchemeId>>
```
Comment (Carq) [inline src/HE.Investment.AHP.Domain/Scheme/CommandHandlers/CreateSchemeCommandHandler.cs:9]: `CreateSchemeCommand` -> `StartNewSchema` IMHO it is more domain friendly name.

File: src/HE.Investment.AHP.Domain/Scheme/Entities/Tenure.cs [hunk 4]
```diff
@@ -0,0 +1,12 @@
+namespace HE.Investment.AHP.Domain.Scheme.Entities;
+
+public enum Tenure{
+    Undefined =0,
```
Comment (Carq) [inline src/HE.Investment.AHP.Domain/Scheme/Entities/Tenure.cs:4]: Stylecop analysers does not work?

**Review by Carq — APPROVED** [review]

File: src/HE.Investment.AHP.Domain/Scheme/CommandHandlers/CreateSchemeCommandHandler.cs (hunk 3 above)
Comment (Antek012) [reply src/HE.Investment.AHP.Domain/Scheme/CommandHandlers/CreateSchemeCommandHandler.cs:9]: fix in next PR

File: src/HE.Investment.AHP.Domain/Scheme/Entities/Tenure.cs (hunk 4 above)
Comment (Antek012) [reply src/HE.Investment.AHP.Domain/Scheme/Entities/Tenure.cs:4]: fix in next PR

File: src/HE.Investment.AHP.Domain/Scheme/CommandHandlers/CreateSchemeCommandHandler.cs (hunk 2 above)
Comment (Antek012) [reply src/HE.Investment.AHP.Domain/Scheme/CommandHandlers/CreateSchemeCommandHandler.cs:22]: Yes it is.

File: src/HE.Investment.AHP.Domain/DomainExceptionHandler.cs (hunk 1 above)
Comment (Antek012) [reply src/HE.Investment.AHP.Domain/DomainExceptionHandler.cs:26]: fix in next PR

**Review by jjankowski87 — APPROVED** [review]

**Pull request merged**